### ISSUE TITLE
[pose-generation][admission] Centralize current invalid-reason and margin derivation without changing validity

### ISSUE THREAD

**Issue by JanDuchscherer104** [JanDuchscherer104/ARIA-NBV#230]
## Outcome

Create one fail-closed owner for deriving the **existing** candidate invalid-reason bitset, primary reason, and named diagnostic margins from current cumulative rule masks and diagnostics.

This salvages the useful reason-precedence and signed-margin logic explored in PR #217 without introducing its public `AdmissionEvidence`/`CriterionEvidence` hierarchy or changing candidate validity.

Related: #54, #72, #73.

## Problem

Candidate-generation rules own hard validity, while rollout trace, inspection, reporting, and persistence need consistent answers to:

- which rule first invalidated a row;
- which additional hard diagnostic reasons also apply;
- which reason wins for one primary reporting label;
- which margin name/unit/sign convention belongs to each rule.

If those projections are independently reconstructed, preflight, plots, Zarr, and CLI reporting can disagree even when `mask_valid` is identical.

## Scope and ownership

This issue is a **diagnostic projection cleanup**, not an admission redesign.

1. `pose_generation` remains the owner of rule order and the numerical masks/diagnostics emitted by current rules.
2. The lowest shared rollout owner derives storage/reporting invalidity from those existing outputs.
3. If only one current consumer derives the bitset/reason, keep the helper local; extract a shared module only after the census proves multiple active consumers.

Start by locating every current implementation in:

```text
aria_nbv/aria_nbv/rollouts/trace.py
aria_nbv/aria_nbv/rollouts/zarr_store.py
aria_nbv/aria_nbv/rollouts/inspection.py
aria_nbv/aria_nbv/rollouts/read_model.py
```

## Minimal algorithm to salvage

For ordered cumulative rule masks `M_k`, derive a local failure only where a row was valid before the rule and invalid afterward:

```python
previous = torch.ones_like(valid_mask)
for rule_name, current in cumulative_masks.items():
    failed_here = previous & ~current
    bitset[failed_here] |= reason_bit(rule_name)
    previous = current
```

Then add current hard diagnostics such as non-finite poses or explicit collision diagnostics without erasing earlier reasons. Derive one primary reason from a single documented priority sequence.

A bounded function may look like:

```python
def derive_candidate_invalidity(
    *,
    valid_mask: torch.Tensor,
    cumulative_rule_masks: Mapping[str, torch.Tensor],
    diagnostic_rejections: Mapping[str, torch.Tensor],
    pose_nonfinite: torch.Tensor,
) -> tuple[torch.Tensor, torch.Tensor]:
    """Return existing reason bitsets and primary reason codes over N."""
```

Use the existing reason-code dictionary and persisted IDs. Do not create a new revision graph merely to move the calculation.

## Margin contract

Inventory existing continuous diagnostics and document, at their numerical owner:

```text
name
unit
applicability/evaluation meaning
sign convention
unavailable representation
```

Use one convention where possible:

```text
positive = inside admissible set
zero     = boundary
negative = violation
```

Do not manufacture a margin where the current rule has no meaningful continuous quantity. Missing/uncomputed remains unavailable, never measured zero.

## Implementation instructions

1. Freeze current rule order, reason IDs, primary precedence, and persisted outputs with characterization tests.
2. Implement one pure derivation over already materialized masks at the current storage/inspection boundary; do not add synchronization to the hot generation path.
3. Migrate all duplicate derivations to the owner in one PR and delete the duplicates.
4. Preserve current `CandidateContext`, `CandidateSamplingResult`, mask mutation, rule execution, reason IDs, and Zarr arrays exactly.
5. Expose only the smallest typed helper needed by demonstrated rollout consumers.
6. Add explicit failure for misaligned shapes/dtypes rather than silently skipping a current-version array.
7. Keep legacy readers permissive only where their schema explicitly lacks a field; distinguish legacy missing from malformed current data.

## Acceptance criteria

- [ ] One implementation owns invalid-reason bitset and primary-reason derivation for current rollout evidence.
- [ ] Rule-local failures are derived from cumulative-mask transitions, not from final invalidity alone.
- [ ] Additional reasons are retained in the bitset while primary precedence is deterministic and documented.
- [ ] Current reason IDs, arrays, bytes, and final `mask_valid` remain unchanged.
- [ ] Existing margins have documented units, applicability, sign, and missing semantics.
- [ ] Missing evidence is not converted to zero.
- [ ] Misaligned current-version masks fail with an actionable error.
- [ ] All-valid, all-invalid, multi-reason, unavailable-diagnostic, and non-finite-pose fixtures pass.
- [ ] #54 and #73 consume the same reason/margin projection rather than reconstructing it.
- [ ] No host/device transfer or tensor reduction is added inside candidate-generation hot loops.

## Explicit exclusions

Do **not** add in this issue:

- a public admission pipeline or criterion plugin registry;
- `CandidateSet`, `CandidateRequest`, or new result DTOs;
- criterion/source-role revision graphs;
- mutation receipts or hashes;
- new feasibility rules from #72;
- persistence-schema changes;
- changes to rule order, thresholds, masks, or production profiles.

## Historical source

PR #217's `derive_invalid_reason_evidence()` and uniform margin convention are useful design evidence. Its broader immutable candidate/admission architecture is superseded.
